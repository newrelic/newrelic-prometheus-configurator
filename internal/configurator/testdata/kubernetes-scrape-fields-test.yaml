kubernetes:
  jobs:
    - job_name_prefix: default
      # prometheus scrape_config fields
      proxy_url: http://proxy.url
      body_size_limit: 100MiB
      sample_limit: 2000
      target_limit: 2000
      label_limit: 2000
      label_name_length_limit: 2000
      label_value_length_limit: 2000
      scrape_interval: 15s
      scrape_timeout: 15s
      tls_config:
        insecure_skip_verify: true
        ca_file: /path/to/ca.crt
        key_file: /path/to/key.crt
        cert_file: /path/to/cert.crt
        server_name: server.name
        min_version: TLS12
      authorization:
        type: Bearer
        # Sets the credentials. It is mutually exclusive with
        # `credentials_file`, we add both for testing purposes
        credentials: "fancy-credentials"
      # Kubernetes job specific configuration
      target_discovery:
        pod: true
        endpoints: true
        additional_config:
          namespaces:
            own_namespace: true
            names:
              - n1
          selectors:
            - role: pod
              label: label-name

newrelic_remote_write:
  license_key: nrLicenseKey
