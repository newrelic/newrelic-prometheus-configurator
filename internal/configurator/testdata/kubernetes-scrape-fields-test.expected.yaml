scrape_configs:
  - job_name: kubernetes-job-pod
    # scrape_job config
    proxy_url: http://proxy.url
    body_size_limit: 100MiB
    sample_limit: 2000
    target_limit: 2000
    label_limit: 2000
    label_name_length_limit: 2000
    label_value_length_limit: 2000
    scrape_interval: 15s
    scrape_timeout: 15s
    tls_config:
      insecure_skip_verify: true
      ca_file: /path/to/ca.crt
      key_file: /path/to/key.crt
      cert_file: /path/to/cert.crt
      server_name: server.name
      min_version: TLS12
    authorization:
      type: Bearer
      credentials: "fancy-credentials"

    kubernetes_sd_configs:
      - role: pod
        namespaces:
          own_namespace: true
          names:
            - n1
        selectors:
          - role: pod
            label: label-name

    relabel_configs:
     # Pod builder configs
      - source_labels: [__meta_kubernetes_pod_phase]
        regex: Pending|Succeeded|Failed|Completed
        action: drop
      - source_labels: [__meta_kubernetes_pod_annotation_prometheus_io_scheme]
        action: replace
        regex: (https?)
        target_label: __scheme__
      - source_labels: [__meta_kubernetes_pod_annotation_prometheus_io_path]
        action: replace
        target_label: __metrics_path__
        regex: (.+)
      - source_labels: [__address__, __meta_kubernetes_pod_annotation_prometheus_io_port]
        action: replace
        regex: (.+?)(?::\d+)?;(\d+)
        replacement: $1:$2
        target_label: __address__
      - action: labelmap
        regex: __meta_kubernetes_pod_annotation_prometheus_io_param_(.+)
        replacement: __param_$1
      - action: labelmap
        regex: __meta_kubernetes_pod_label_(.+)
      - source_labels: [__meta_kubernetes_namespace]
        action: replace
        target_label: namespace
      - source_labels: [__meta_kubernetes_pod_node_name]
        action: replace
        target_label: node
      - source_labels: [__meta_kubernetes_pod_name]
        action: replace
        target_label: pod

  - job_name: kubernetes-job-endpoints
    # scrape_job config
    proxy_url: http://proxy.url
    body_size_limit: 100MiB
    sample_limit: 2000
    target_limit: 2000
    label_limit: 2000
    label_name_length_limit: 2000
    label_value_length_limit: 2000
    scrape_interval: 15s
    scrape_timeout: 15s
    tls_config:
      insecure_skip_verify: true
      ca_file: /path/to/ca.crt
      key_file: /path/to/key.crt
      cert_file: /path/to/cert.crt
      server_name: server.name
      min_version: TLS12
    authorization:
      type: Bearer
      credentials: "fancy-credentials"

    kubernetes_sd_configs:
      - role: endpoints
        namespaces:
          own_namespace: true
          names:
            - n1
        selectors:
          - role: pod
            label: label-name

    relabel_configs:
    # Endpoints builder configs
      - source_labels: [ __meta_kubernetes_pod_phase ]
        action: drop
        regex: Succeeded|Failed|Completed
      - source_labels: [__meta_kubernetes_service_annotation_prometheus_io_scheme]
        action: replace
        target_label: __scheme__
        regex: (https?)
      - source_labels: [__meta_kubernetes_service_annotation_prometheus_io_path]
        action: replace
        target_label: __metrics_path__
        regex: (.+)
      - source_labels: [ __address__, __meta_kubernetes_service_annotation_prometheus_io_port ]
        action: replace
        regex: (.+?)(?::\d+)?;(\d+)
        target_label: __address__
        replacement: $1:$2
      - action: labelmap
        regex: __meta_kubernetes_service_annotation_prometheus_io_param_(.+)
        replacement: __param_$1
      - action: labelmap
        regex: __meta_kubernetes_service_label_(.+)
      - source_labels: [__meta_kubernetes_namespace]
        action: replace
        target_label: namespace
      - source_labels: [__meta_kubernetes_service_name]
        action: replace
        target_label: service
      - source_labels: [ __meta_kubernetes_endpoint_node_name, __meta_kubernetes_pod_node_name ]
        separator: ;
        action: replace
        regex: ".*;(.+)|(.+);"
        replacement: "$1$2"
        target_label: node

remote_write:
  - name: newrelic_rw
    url: https://metric-api.newrelic.com/prometheus/v1/write?collector_name=prometheus-agent
    authorization:
      credentials: nrLicenseKey
